### DISCUSSION TITLE
Filter by current repository

### DISCUSSION THREAD

**Discussion by joshmedeski** [dlvhdr/gh-dash#309] [Ideas]
I'd like to be able to open up gh-dash while on a repository, and it only show me items specific to the project I'm on.

What do we need to do to make this happen? I'm open to do some development.

**Comment by dlvhdr** [comment]
I'd love that :) We can hop on a voice call and I explore together what needs to be done if you're down

**Comment by fpohtmeh** [comment]
Any update here?

**Reply by dlvhdr** [reply]
It's now supported to have gh-dash read a .gh-dash.yml file at the root of the repo. Does that meet your needs?

**Reply by fpohtmeh** [reply]
Wow, so it works. Great!

How about adding a placeholder for the current directory repo?
```
issuesSections:
  - title: Open
    filters: author:@me repo:@current_repo
```
-> `@current_repo` 
So I shouldn't specify it directly and use same config in many repos

**Comment by sideshowbarker** [comment]
At https://github.com/dlvhdr/gh-dash/pull/526 I took a shot at putting together a patch for implementing what’s originally proposed here.

The support for putting a `.gh-dash.yml` config file in the current directory is absolutely great, but especially if you have need to go through and look at PRs/Issues for _a lot_ of repos (which is a real-world use-case for me, with the @w3c org, where we have literally 1.1 _thousand_ repos — https://github.com/orgs/w3c/repositories?type=source), then it seems preferable for that case to have the behavior be that `gh-dash` by default “smartly” detects you’re in a clone directory, and auto-filters down to the GitHub remote for that clone.

So with that patch, the default behavior — without you needing to do anything — would be that `gh-dash` automatically filters your dashboard down to PRs and Issues for only those of the GitHub repo for the clone directory you’re in. And if you do want to view your global dashboard while you’re in a clone directory, then there’s a key you can hit to toggle back and forth to/from that view.

All that with the exception of: If a section in your config has a`filters` value with an explicit `repo:foo`, then `gh-dash` respects that explicit repo name — and for that section, doesn’t try to “smartly” filter anything for that section down to the current clone/remote.

So all that’s what the patch in the https://github.com/dlvhdr/gh-dash/pull/526 PR attempts to do.

And if you don’t want that automatic smart-filtering behavior to happen by default at startup, there’s a `smartFilteringAtLaunch` config key that you can set to `false` to prevent it. And then if/when you _do_ want to see the filtered-to-current-remote view — but just not by default at startup — than after startup, whenever you might want to, you can use the toggle key to switch over to that view.

> How about adding a placeholder for the current directory repo?
> 
> ```
> issuesSections:
>   - title: Open
>     filters: author:@me repo:@current_repo
> ```

With the patch in https://github.com/dlvhdr/gh-dash/pull/526, you wouldn’t need to do that — nor make any other changes to any of your existing config. Instead, `repo:@current_repo` would basically be the implicit default — though you’d additionally then also be able to toggle back and forth between the `repo:@current_repo` view and the “not just `repo:@current_repo`” view as much as you want, whenever you want.
